### ISSUE TITLE
docs: consider switching to sphinx-book-theme

### ISSUE THREAD

**Issue by ltalirz** [aiidateam/aiida-core#4441]
As shown by @chrisjsewell , the sphinx-book-theme offers a couple of nices features, including:
 * suggesting edits
 * opening issue for the page one is looking at
 * downloading a page as PDF

<img src="https://user-images.githubusercontent.com/1053245/96010438-12bdef80-0e42-11eb-9987-02c43ddae8ea.png" height=150>
<img src="https://user-images.githubusercontent.com/1053245/96010579-3e40da00-0e42-11eb-9a72-ced328c5f0b0.png" height=150>

I'm interested particularly in lowering the barrier for users to give feedback on the documentation - part of it is perhaps just textual: we currently have an "edit this page" button
![image](https://user-images.githubusercontent.com/1053245/96010684-63354d00-0e42-11eb-851d-949394ae1e48.png)
but I think "suggest edit" is actually what we want people to see.

I just discussed with @chrisjsewell and he suggested to wait a few weeks until there is a stable release.

It should mainly just be replacing `pydata-sphinx-theme` with `sphinx-book-theme`

**Comment by chrisjsewell** [comment]
Yep FYI: `sphinx-book-theme` literally inherits from `pydata-sphinx-theme`, so it should be too difficult to change over in a test branch see what the docs look like.
We've got a few imminent improvements in the pipeline, which is why I suggested to wait just a little bit

**Comment by chrisjsewell** [comment]
Note, sphinx-book-theme is just about there now with the improvements I mentioned (by moi as one of its maintainers): https://sphinx-book-theme.readthedocs.io/en/latest/ (e.g. expandable sections in the left ToC), and as mentioned in #5006, @mbercx has already taken the plunge to use it in https://aiida-tutorials.readthedocs.io/en/tutorial-2021-intro/ 😄

**Comment by mbercx** [comment]
> @mbercx has already taken the plunge to use it in https://aiida-tutorials.readthedocs.io/en/tutorial-2021-intro/ 😄

And a deep dive it was 😅

**Comment by ltalirz** [comment]
@chrisjsewell Is there any reason to wait more for this or is it up for grabs?

**Comment by chrisjsewell** [comment]
No go for it, all the options are here: https://sphinx-book-theme.readthedocs.io/en/latest/customize/index.html#theme-options

and also switch sphinx-panels to sphinx-design: https://sphinx-design.readthedocs.io/en/sbt-theme/get_started.html#migrating-from-sphinx-panels

**Comment by ltalirz** [comment]
@chrisjsewell will take care of this once sphinx-book-theme has further consolidated, see https://github.com/aiidateam/aiida-core/pull/5368
